catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -44,72 +40,21 @@ class WorksController @Inject()(@Flag("api.prefix") apiPrefix: String,
     .items(new StringProperty()._enum(WorksIncludes.recognisedIncludes.asJava))
 
   prefix(apiPrefix) {
+    setupResultListEnpoint("/works")
 
-    val endpointSuffix = "/works"
+    setupSingleWorkEnpoint("/works/:id")
+  }
 
+  private def setupResultListEnpoint(endpointSuffix: String): Unit = {
     getWithDoc(endpointSuffix) { doc =>
-      doc
-        .summary(endpointSuffix)
-        .description("Returns a paginated list of works")
-        .tag("Works")
-        .responseWith[DisplayResultList](200, "ResultList[Work]")
-        .responseWith[DisplayError](400, "Bad Request Error")
-        .responseWith[DisplayError](404, "Not Found Error")
-        .responseWith[DisplayError](500, "Internal Server Error")
-        .queryParam[Int](
-          "page",
-          "The page to return from the result list",
-          required = false)
-        .queryParam[Int](
-          "pageSize",
-          "The number of works to return per page (default: 10)",
-          required = false)
-        .queryParam[String](
-          "query",
-          """Full-text search query, which will OR supplied terms by default.
-            |
-            |The following special characters can be used to change the search behaviour:
-            |
-            |- \+ signifies AND operation
-            |- | signifies OR operation
-            |- \- negates a single token
-            |- " wraps a number of tokens to signify a phrase for searching
-            |- \* at the end of a term signifies a prefix query
-            |- ( and ) signify precedence
-            |- ~N after a word signifies edit distance (fuzziness)
-            |- ~N after a phrase signifies slop amount
-            |
-            |To search for any of these special characters, they should be escaped with \.""".stripMargin,
-          required = false
-        )
-        .parameter(includesSwaggerParam)
-    // Deliberately undocumented: we have an 'index' query param that
-    // allows the user to pick which Elasticsearch index to use.  This is
-    // useful for us to try out transformer changes, different index
-    // weighting, etc., but we don't want to advertise its existence
-    // in the public docs.
-
+      setupResultListSwaggerDocs(endpointSuffix, doc)
```

Comment:
Let’s not do this now, but it occurs to me you could pass `DisplayResultList` as a type parameter – thus allowing us to reuse this when we fork the controllers.

Nice!

Reply:
(Although we may end up changing the behaviour, so need to copy it. Oh well, was nice thought.)